Fixed a bunch of outdated pad styles

evanshultz opened the pull request:
**DO NOT MERGE YET AS ALL ITEMS BELOW ARE NOT CHECKED OFF**

This is a big one, so grab a cup of coffee and sit down in a comfy chair. Or... you can trust me and merge with a quick review. :)

I set out to address footprints listed at https://github.com/KiCad/kicad-footprints/issues/266, and I found a few more with outdated pad designs so I fixed them up too.

Changed multiple overlapping pads to a single custom pad:
- [ ]  Inductor_SMD/L_Coilcraft_LPS4018 (also moved silkscreen from under part body)
- [ ]  Inductor_SMD/L_Wuerth_WE-TPC-3816
- [ ]  Package_DFN_QFN/Infineon_MLPQ-16-14-1EP_4x4mm_P0.5mm
- [ ]  Package_DFN_QFN/Infineon_MLPQ-40-32-1EP_7x7mm_P0.5mm
- [ ]  Package_DFN_QFN/Infineon_MLPQ-48-1EP_7x7mm_P0.5mm_Pad5.15x5.15mm (rename?)
- [ ]  Package_DFN_QFN/Infineon_MLPQ-48-1EP_7x7mm_P0.5mm_Pad5.55x5.55mm (rename?)
- [ ]  Package_DFN_QFN/Infineon_PQFN-22-15-4EP_6x5mm_P0.65mm
- [ ]  Package_DFN_QFN/Infineon_PQFN-44-31-5EP_7x7mm_P0.5mm
- [ ]  Package_DFN_QFN/Linear_DE14MA (overlapping pins had different pin numbers(!) and updated copper and fixed 3D model name)
- [ ]  Package_DFN_QFN/ST_UFQFPN-20_3x3mm_P0.5mm
- [ ]  Package_TO_SOT_SMD/HVSOF5 (and fix pad X position)
- [ ]  Package_TO_SOT_SMD/LFPAK33
- [ ]  Package_TO_SOT_SMD/LFPAK56 (I didn't touch the other LFPAK56 because they're non-standard and perhaps should be removed?)
- [ ]  Package_TO_SOT_SMD/PQFN_8x8 (this needed a few dimensional fixes too)
- [ ]  Package_TO_SOT_SMD/SOT-89-3 (no referenced datasheet so picked the closest one and updated the footprint to match)
- [ ]  Package_TO_SOT_SMD/SOT-89-3_HandSoldering
- [ ]  Package_TO_SOT_SMD/SOT-89-5
- [ ]  Package_TO_SOT_SMD/SOT-89-5_HandSoldering
- [ ]  Package_TO_SOT_SMD/TDSON-8-1 (nearly all dimensions were wrong so this got fixed too)
- [ ]  Package_TO_SOT_SMD/Vishay_PowerPAK_SC70-6L_Single

Removed pin number from paste pads:
- [ ]  Package_TO_SOT_SMD/ATPAK-2
- [ ]  Package_TO_SOT_SMD/ROHM_HRP7

Merged multiple adjacent pads into a single one:
- [ ]  Package_DFN_QFN/DFN-S-8-1EP_6x5mm_P1.27mm
- [ ]  Package_DFN_QFN/Linear_UGK52_QFN-46-52
- [ ]  Package_DFN_QFN/Microchip_8E-16
- [ ]  Package_DFN_QFN/Microsemi_QFN-40-32-2EP_6x8mm_P0.5mm (also fixed line widths)
- [ ]  Package_DFN_QFN/Mini-Circuits_DL805 (and removed silk to mark pin 1)
- [ ]  Package_DFN_QFN/Texas_RUM0016A
- [ ]  Package_DFN_QFN/Texas_R_PUQFN-N12
- [ ]  Package_DFN_QFN/Texas_VQFN-RGR-20
- [ ]  Package_DFN_QFN/Texas_VQFN-RGW-20
- [ ]  Package_DFN_QFN/Texas_WQFN-MR-100_3x3-DapStencil (and updated silk pin 1 mark)
- [ ]  Package_DFN_QFN/Texas_WQFN-MR-100_ThermalVias_3x3-DapStencil (and updated silk pin 1 mark and added missing vias from doc and rename to put _ThermalVias at the end?)
- [ ]  Package_DFN_QFN/UDFN-4_1x1mm_P0.65mm (and updated silk and moved courtyard to 0.25mm clearance and added datasheet)
- [ ]  Package_TO_SOT_SMD/Infineon_PG-HSOF-8-1 (basically a total do-over)
- [ ]  Package_TO_SOT_SMD/Infineon_PG-HSOF-8-1_ThermalVias (basically a total do-over)
- [ ]  Package_TO_SOT_SMD/Infineon_PG-HSOF-8-2 (new footprint)
- [ ]  Package_TO_SOT_SMD/Infineon_PG-HSOF-8-2_ThermalVias (new footprint)
- [ ]  Package_TO_SOT_SMD/Infineon_PG-HSOF-8-2_ThermalVias2 (new footprint)

Other changes:
- [ ]  Package_TO_SOT_SMD/TDSON-8-1_HandSoldering: Removed since the X dimension was +/-3.5mm and the fixed reflow one is +/-3.325mm

I left alone QFNs with odd-shaped corner pads and/or D-shaped pads since I believe Rene's fancy generator script can (or could) make them.

Dimensioned drawings and screenshots, where relevant, to follow.

------------

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the footprint(s) you are contributing
- [ ] An example screenshot image is very helpful 
- [ ] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

evanshultz commented:
Obviously a ton of Travis errors, mostly because these are old footprints. I'll work through them.

evanshultz commented:
Coilcraft LPS4018

https://www.coilcraft.com/pdfs/lps4018.pdf

![image](https://user-images.githubusercontent.com/1936989/46229671-3f664380-c31b-11e8-878c-677915db4de7.png)

![image](https://user-images.githubusercontent.com/1936989/46229663-37a69f00-c31b-11e8-9c7e-7be06e469b32.png)

KSU appears to only import the first primitive of a custom pad so double the numbers on the left side for pad heights.

evanshultz commented:
http://www.analog.com/media/en/package-pcb-resources/package/pkg_pdf/ltc-legacy-dfn/05081731_C_DE14MA.pdf

![image](https://user-images.githubusercontent.com/1936989/46229772-90763780-c31b-11e8-81b7-1869c5dfa7f1.png)

The dimensioned drawing isn't even worth doing since what I changed are pads 4 and 9 and KSU doesn't import the custom pad shapes (yet?).

evanshultz commented:
https://www.rohm.com/datasheet/BU52014HFV/bu52004gul-e

![image](https://user-images.githubusercontent.com/1936989/46230059-3d50b480-c31c-11e8-8245-c5867f46d3b5.png)

![image](https://user-images.githubusercontent.com/1936989/46233147-7aba3f80-c326-11e8-809f-a05d0358d8c2.png)

I fixed the courtyard after dimensioning this.

evanshultz commented:
https://www.fairchildsemi.com/datasheets/FD/FDMT80080DC.pdf

![image](https://user-images.githubusercontent.com/1936989/46233758-bd7d1700-c328-11e8-8358-8db8a65df593.png)

![image](https://user-images.githubusercontent.com/1936989/46233975-a25ed700-c329-11e8-90c4-5cc49d6ce008.png)

I fixed the courtyard after dimensioning.

evanshultz commented:
http://ww1.microchip.com/downloads/en/DeviceDoc/3L_SOT-89_MB_C04-029C.pdf

![image](https://user-images.githubusercontent.com/1936989/46234096-04b7d780-c32a-11e8-94c2-70cf4fbcaf53.png)

![image](https://user-images.githubusercontent.com/1936989/46234421-1b126300-c32b-11e8-9108-7d542527bed1.png)

I kept the existing body width of 5mm even though this Microchip datasheet shows 4.5mm.

evanshultz commented:
https://www.infineon.com/cms/en/product/packages/PG-TDSON/PG-TDSON-8-1/

![image](https://user-images.githubusercontent.com/1936989/46234467-485f1100-c32b-11e8-973e-87fa496576ae.png)

![image](https://user-images.githubusercontent.com/1936989/46234732-7bee6b00-c32c-11e8-9995-e81797c142b6.png)

evanshultz commented:
http://ww1.microchip.com/downloads/en/DeviceDoc/MIC550x-300mA-Single-Output-LDO-in-Small-Packages-DS20006006A.pdf

![image](https://user-images.githubusercontent.com/1936989/46234811-c374f700-c32c-11e8-952c-87551656df7b.png)

![image](https://user-images.githubusercontent.com/1936989/46235029-5746c300-c32d-11e8-933d-2dff0380f3a5.png)

I'm not sure what is the correct courtyard spacing for this. Should it be 0.25mm like other DFNs? 1mm like BGAs? Either way, it's not equal on all sides so something is wrong.

evanshultz commented:
https://www.infineon.com/cms/en/product/packages/PG-HSOF/PG-HSOF-8-1/

https://www.infineon.com/cms/en/product/packages/PG-HSOF/PG-HSOF-8-2/

![image](https://user-images.githubusercontent.com/1936989/46235130-c1f7fe80-c32d-11e8-8e87-dc0eab43156a.png)

![image](https://user-images.githubusercontent.com/1936989/46238278-325b4b80-c33f-11e8-82a1-49eab7350950.png)

![image](https://user-images.githubusercontent.com/1936989/46238351-0db3a380-c340-11e8-9f86-0e16831ce07c.png)

evanshultz commented:
That's all the footprints that need dimensioned drawings, I think. Next I'll show screenshots for all footprints (save the Wuerth one that isn't finished) in the order they're listed above unless the changes was simply making one big pads instead of a matrix of pads.

![image](https://user-images.githubusercontent.com/1936989/46238376-553a2f80-c340-11e8-943e-be55445cf7cb.png)

![image](https://user-images.githubusercontent.com/1936989/46238397-861a6480-c340-11e8-82cf-a6bad0b6ffce.png)
![image](https://user-images.githubusercontent.com/1936989/46238400-8d417280-c340-11e8-9946-e3cc84d56d57.png)
![image](https://user-images.githubusercontent.com/1936989/46238402-98949e00-c340-11e8-9643-8c22d93dd820.png)

![image](https://user-images.githubusercontent.com/1936989/46238409-a518f680-c340-11e8-9112-34a446751e28.png)

![image](https://user-images.githubusercontent.com/1936989/46238417-bb26b700-c340-11e8-9d6e-90f68a078083.png)

![image](https://user-images.githubusercontent.com/1936989/46238426-cc6fc380-c340-11e8-8667-3013d998d601.png)

![image](https://user-images.githubusercontent.com/1936989/46238435-e6a9a180-c340-11e8-86b9-527a8b1cdf9a.png)

![image](https://user-images.githubusercontent.com/1936989/46238439-f45f2700-c340-11e8-938c-9e3095f8ef71.png)

![image](https://user-images.githubusercontent.com/1936989/46238445-0b9e1480-c341-11e8-9c08-1ae64c591dc4.png)
![image](https://user-images.githubusercontent.com/1936989/46238449-1e184e00-c341-11e8-8288-76443385fc3a.png)

![image](https://user-images.githubusercontent.com/1936989/46238457-2a9ca680-c341-11e8-99e2-07d254083309.png)

![image](https://user-images.githubusercontent.com/1936989/46238464-443dee00-c341-11e8-8a57-9591e48ae4c3.png)

![image](https://user-images.githubusercontent.com/1936989/46238481-5ddf3580-c341-11e8-936c-21edbce9567d.png)

![image](https://user-images.githubusercontent.com/1936989/46238518-9848d280-c341-11e8-8be0-300aee4639b2.png)

![image](https://user-images.githubusercontent.com/1936989/46238543-c0383600-c341-11e8-9c8d-1cfc5599b873.png)

![image](https://user-images.githubusercontent.com/1936989/46238550-cfb77f00-c341-11e8-975d-01a0fe3c13e2.png)

![image](https://user-images.githubusercontent.com/1936989/46238556-de9e3180-c341-11e8-96be-067f6094373a.png)

evanshultz commented:
That leaves the following questions
1. Should `Package_DFN_QFN/Infineon_MLPQ-48-1EP_7x7mm_P0.5mm_Pad5.x5x5.x5mm` footprints have the `_Pad*` removed from their 3D model name? I think so.
1. Should the same footprints use a different nomenclature? I don't know why they got merged this way but the `_Pad*` should be `_EP*`. Change it?
1. Remove LFPAK56 with vias since they're not from the datasheet and seem like a specific implementation? All vias are way outside the footprint boundary. It seems to me this should be done on the board or a local library but not in our official library.
1. Should SOT-89 footprints shrink the body size to match the datasheet? I chose one with the biggest pads that were similar (but still a bit smaller) than the footprint in our lib which had no datasheet.
1. Rename `Texas_WQFN-MR-100_*3x3-DapStencil` to something better? Perhaps just `Texas_S-PWQFN-N100` or `Texas_RKM-100` (we should really figure out how we are going to name these weird TI footprints) and of course the `_ThermalVias` version?
1. Courtyard clearance for `UDFN-4_1x1mm_P0.65mm`?

I wrote a short script to search libraries for the footprint used so that can help us track down some of the potential impact of the items above.

evanshultz commented:
I just pushed the last footprint I'm aware of that belongs in this group of footprints.
![image](https://user-images.githubusercontent.com/1936989/46305468-5d73b400-c566-11e8-946d-1353dec2409f.png)

![image](https://user-images.githubusercontent.com/1936989/46305457-5351b580-c566-11e8-912b-d4cff3f067ee.png)

This is now ready for review. Please see the questions above.

evanshultz commented:
Just commited changes to the Package_SO lib. Didn't notice those before. In general I just kept them the same as before only merging pads. The list in the first comment is updated.

![image](https://user-images.githubusercontent.com/1936989/46693464-b07ce500-cbbe-11e8-8e20-1b3d59177e92.png)
![image](https://user-images.githubusercontent.com/1936989/46693505-cf7b7700-cbbe-11e8-9261-72202a78cab9.png)
![image](https://user-images.githubusercontent.com/1936989/46693516-d73b1b80-cbbe-11e8-8e57-997138413265.png)

Misca1234 commented:
@KiCad/librarians 
> This is a big one, so grab a cup of coffee and sit down in a comfy chair. 
> Or... you can trust me and merge with a quick review. :)

Anyone oppose to trust him ?

poeschlr commented:
The main thing i would suggest to check is that all pads have zone connect set to solid instead of none (for polygon pads the default is none so it could easily happen. There sadly is no option to select thermals for polygon pads.)

evanshultz commented:
@poeschlr 
So what you write only applies, to polygon pads, right?

I disagree with having a solid connection on all pads because that will make soldering difficult, but if the only choices are Solid and None I understand why Solid must be chosen. Sigh.

Please confirm the below and I'll update.

@Misca1234 
That should work for footprints with only custom pads.

However, most of these footprints have only a few custom pads and so I believe we will then use the pad-specific connection setting seen here:
![image](https://user-images.githubusercontent.com/1936989/46965390-92503280-d05f-11e8-9fa1-7aadc75f7b56.png)

We would not want Solid connection everywhere if it doesn't have to be, so we can do it per-pad on many footprints.

evanshultz commented:
Oops. I didn't mean to submit that comment above yet.

During some quick experimentation, the per-footprint settings appear to have no effect if there are only custom pads. Only the per-pad settings seem to have any effect.

@poeschlr 
So this means every single custom pad in this PR needs the pad-specific zone connection setting changed to Solid, right? And this should be the default for all custom pads in the library with the current behavior of KiCad?

evanshultz commented:
@poeschlr @jkriege2 @Misca1234 @antoniovazquezblanco @DanSGiesbrecht 
And before merging, I would like to answer the questions posed at https://github.com/KiCad/kicad-footprints/pull/981#issuecomment-425599992.

evanshultz commented:
Any and all: please see and respond to the questions mentioned above.

@poeschlr 
Please confirm you want the per-pad zone connection set to Solid as I described above.

evanshultz commented:
There are a number of open items above that I'd like help to answer.

@poeschlr 
Can you please confirm the change above? I want to make sure I'm understanding.

evanshultz commented:
@poeschlr 
Another poke.

evanshultz commented:
@poeschlr 
Please review my question above about zone connection.

poeschlr commented:
For custom pads we only have the option solid. (And none but i feel that one will do more harm than good.) This is a known restriction of custom pads and the reason why v6 will get a chamfered pad.

evanshultz commented:
Thanks for answering. All custom pads should be set to Solid connection now except for the four that GitHub has trouble figuring out (they are unchecked above).

That leaves the questions I posed at https://github.com/KiCad/kicad-footprints/pull/981#issuecomment-425599992.

evanshultz commented:
@poeschlr (and anyone else interested in contributing)
Can you please look at the questions I raised at https://github.com/KiCad/kicad-footprints/pull/981#issuecomment-425599992? I'd like to get this PR ready for a review and those need answer first. Thanks!

poeschlr commented:
To 1: Yes. (but the answer to 2 makes this question redundant.)
To 2: Without looking at the footprint i would still guess that yes it should be EP.
To 3: yes the current thermal vias version seems strange. I would replace it with a better one that does not increase the pad area.
To 4: Yes any footprint must match the dimensions given in the datasheet assigned to it (exception: If we use industry standards to create the footprint. Then the input to the calculation must match the dimensions given in the datasheet.)
To 5: Texas_S-PWQFN-N100 would fit other footprints. If it has an exposed pad then the size of it should also be included.

evanshultz commented:
Thanks for answering!

1 & 2. The 5.15x5.15mm datasheet actually shows a 5.55x5.55mm EP, so I removed the 5.15 one and changed `Pad` to `EP`. No symbols are currently using this footprint, but that's because they were left out of the v5 library migration. While it doesn't depend on this PR in any way, I submitted https://github.com/KiCad/kicad-symbols/pull/1395 to re-add them.
3. Removed.
4. Updated using dimensions from IPC-SM-782 page 96, which also matched both referenced datasheets (2.5x4.5mm).
5. Renamed to `Texas_S-PWQFN-N100_EP5.5x5.5mm`. One symbol was affected and https://github.com/KiCad/kicad-symbols/pull/1394 will fix it. Since the symbol PR above is substantial I broke this one out separately.

What about item 6?

evanshultz commented:
@poeschlr 
Can you check on item 6, referenced above? The courtyard offset was 1mm, but I don't see why it shouldn't be 0.25mm.

poeschlr commented:
If the udfn footprint does not give a specific clearance in the datasheet then i would suggest 0.25mm clearance. (It is not a bga footprint but more like a lga or qfn. both of these classes have 0.25mm clearance set in the IPC standards)

evanshultz commented:
Great. That's what I thought. Courtyard updated:
![image](https://user-images.githubusercontent.com/1936989/51420654-ed0a9600-1b48-11e9-8989-01b2b1e5b32b.png)

So... anything else?

evanshultz commented:
@poeschlr 
Can this be merged now?

evanshultz commented:
Note to self: In addition to the linked PRs above, https://github.com/KiCad/kicad-symbols/pull/1395 is using the renamed Infineon MLPQ-48 footprints.

evanshultz commented:
@poeschlr 
Do you see anything stopping this from being merged?

poeschlr commented:
There seems to be a merge conflict now. Could you look into that?

evanshultz commented:
OK. Branch conflicts fixed and Travis is happy.

I only touched the few QFNs mentioned above. I couldn't know or expect somebody would do so much replacing of existing QFNs/QFNs in the last, gosh, ~4.5 months since I created this PR. I don't see anything that went wrong so I believe this is still good to merge.

evanshultz commented:
@poeschlr 
Be warned that is this is merged at least one symbol PR needs to be merged and there are several other PRs that will then be "unlocked". If you merge this I'll go around to the other PRs and make a note there.

evanshultz commented:
@poeschlr 
You've done most of the answering and advice here. Is this able to be merged?

poeschlr commented:
I am not prepared to merge footprints that have not been checked in detail. I also do not currently have the time (or energy) to check 50 plus footprints in one go. 
Right now i suspect this will require a few weeks of slow and tedious work. (Meaning very low chance of this being done for v5.1 which is likely the source of your frustration.)

For the future please limit manual changes to the footprints to a handful of parts per pull request. Scripted changes for the most part are a bit different with this requirement. But this also only applies if the generated footprints are highly similar. If they are completely different then we can not simply only check one or two extremes and we are back to the same requirement as for manual changes.

---

My reasoning behind the requirement for thorough checks stems from my experience of converting footprints over to the script. I discovered that a few years ago the library team only checked contributions against the KLC not against the datasheet. Nearly all these old parts have slight errors (especially when they include exposed pads.) I do not want to repeat the same mistake and leave the library in a similar state for future librarians.
This is compounded in this case from my knowledge of how bad the polygon pad drafting is in kicad. -> Makes it likely to make subtle mistakes.

evanshultz commented:
I included dimensioned drawings above to hopefully bootstrap the review process. Is there something more contributors can do? (Splitting changes up into multiple PRs is the same amount of work, just distributed.)

evanshultz commented:
Remove the Infineon QFN needed by https://github.com/KiCad/kicad-symbols/pull/1395 and popped it into a PR all by itself at https://github.com/KiCad/kicad-footprints/pull/1414.

myfreescalewebpage commented:
Conflict due to merging of #1459 to be fixed @evanshultz

myfreescalewebpage commented:
@evanshultz looking at the git diff, I'm not sure of the deletion of the two footprints we have discussed...

evanshultz commented:
Here it is in renaming of these files in the web editor:
![image](https://user-images.githubusercontent.com/1936989/53822315-5ccfb700-3f24-11e9-8d70-d826e3ff0bc6.png)

And I checked the files in this branch again and the `DapStencil` one don't exist.

Am I missing something?

myfreescalewebpage commented:
Following the integration of the other PR, the new footprint just simply exist.
In the current PR, we should just see that the DapStencil footprint are removed, not renamed, else you will overwrite the new footprints.

evanshultz commented:
@antoniovazquezblanco 
While the Infineon footprint has been merged, the Wuerth footprint is at https://github.com/KiCad/kicad-footprints/pull/981 and has not been merged.  So I think it's premature to merge this. Once I wrap up a couple things I will break apart #981 for easier merging.

poeschlr commented:
@evanshultz is this still relevant or was it replaced with the other pull requests?

evanshultz commented:
I believe all changes here should be in separate PRs, but I kept this open as a 'meta' bug to track progress.

myfreescalewebpage commented:
@evanshultz I think @poeschlr asks if the PR is not actually fully fixed ? The checkboxes in the first comment are all checked. It's difficult to know the remaining task actually on this.

jneiva08 commented:
@evanshultz i need to make footprint for Wuerth TPC with another size
in the description you have put, "StepUp generated footprint"
How can use StepUp to make?

evanshultz commented:
@jneiva08 
I am no expert but I got help from @easyw at https://github.com/KiCad/kicad-footprints/pull/972. Hopefully that documents my struggles enough for you to follow along.
